docs/build/guides/conventions/cross-contract.mdx

```diff
@@ -0,0 +1,130 @@
+pub struct ContractB;
+
+#[contractimpl]
+impl ContractB {
+    pub fn add_with(env: Env, contract: Address, x: u32, y: u32) -> u32 {
+        let client = contract_a::Client::new(&env, &contract);
+        client.add(&x, &y)
+    }
+}
+```
+
+Although we have access to the Wasm, we still need a contract address because there could be multiple contracts deployed on-chain that use the same underlying code. This can be of importance if, for instance, the function that you need requires access to stored values from other users.
+
+:::tip[Address and Wasm]
+
+Thanks to the Rust bindings of the external contract, we can also use any public enum of the contract as if we would have defined them within our own contract.
+
+```rust
+client::ContractAEnum::SomeField
+```
+
+:::
+
+## Handling responses
+
+In the examples above, we have used `env.invoke_contract` and `client.some_function`. In both cases, if there is an issue with the underlying contract call, the contract will panic. This might be a valid approach, but in some cases we want to catch errors and handle them depending on the outcome. This is to forward a custom error message, or even triggers an alternative code path.
+
+Enter `try_`. By using `env.try_invoke_contract` or `client.try_some_function`, underlying errors won't make the contract panic. Instead, errors will be wrapped and can be handled. For example if we wanted to default to 0 in case of an error:
+
+```rust
+client.try_add(&x, &y).unwrap_or(Ok(0)).unwrap()
+```
+
+Of course, we could have far more complex error handling by leveraging the `match` statement:
+
+```rust
+match client.try_add(&x, &y) {
+    // the contract returned a value
+    Ok(Ok(number)) => todo!("do something with the number returned"),
+    Ok(Err(ConversionError)) => todo!("got a value back, but it wasn't a number like we expected"),
+
+    // the contract errored
+    Err(Ok(Error::AnError)) => todo!("do something when an error occurs that the contract included in its contract spec"),
+    Err(Err(status)) => todo!("do something when an unrecognized error, or system error, occurs"),
+}
+```
+
+## Depending on another contract
+
+Congratulations, now you can effectively leverage the whole Soroban ecosystem and its myriad of smart contracts. There is one last point to discuss before closing up: **dependability**.
+
+As with calling any smart contract, depending on an external smart contract should be done with care. It is advisable to do your own research and analysis on the contract you want to use. As contracts can be updated without their address changing, it is important to pay close attention to any changes in the underlying code.
+
+:::tip[Contract Wasm]
+
+The Wasm can be fetched by using the [Stellar CLI](../../../tools/developer-tools/cli). This can serve as a quick way to (for example) automate a hash check in a continuous integration system. However, such a check would not provide strong on-chain guarantees, but one could build a contract for that!
```

Comment:
```suggestion
The Wasm can be fetched by using the [Stellar CLI](../../../tools/developer-tools/cli/README.mdx). This can serve as a quick way to (for example) automate a hash check in a continuous integration system. However, such a check would not provide strong on-chain guarantees, but one could build a contract for that!
```

I know it feels silly to link to a `directory/README.mdx` file, but it's essentially like linking to an `index.html` page, and it's the best way to get the benefits of linking to the `*.mdx` file for a URL that (in this case) would just end like `/cli`

Reply:
Thanks for the explanation!